src/DIRAC/WorkloadManagementSystem/Agent/test/Test_Agent_PilotLoggingAgent.py

```diff
@@ -0,0 +1,200 @@
+import unittest
+import os
+from contextlib import contextmanager
+from unittest.mock import MagicMock, patch, call
+
+from DIRAC import gLogger, gConfig, S_OK, S_ERROR
+import DIRAC.WorkloadManagementSystem.Agent.PilotLoggingAgent
+from DIRAC.WorkloadManagementSystem.Agent.PilotLoggingAgent import PilotLoggingAgent
+
+
+@contextmanager
+def patch_parent(class_):
+    """
+    Mock the bases.
+    source: https://stackoverflow.com/a/24577389/9893423
+    """
+    yield type(class_.__name__, (MagicMock,), dict(class_.__dict__))
```

Comment:
IIUC what you are doing here is mocking the baseclass of PilotLoggingAgent, right? (`AgentModule`)

There are several examples of this being done in the DIRAC code, e.g. check any of the tests in https://github.com/DIRACGrid/DIRAC/tree/integration/src/DIRAC/WorkloadManagementSystem/Agent/test

Reply:
And in general I think `pytest` is better unit testing framework, largely used in DIRAC code.

Reply:
Moved to pytest